### PR TITLE
[Feature] support snappy stream compression

### PR DESCRIPTION
**Pull request by dirtysalt**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 55 / 63 (87.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_text.cpp|1|5|20.00%|[389, 390, 391, 393]|
|:large_blue_circle:|src/util/compression/stream_compression.cpp|50|54|92.59%|[430, 480, 488, 500]|
|:large_blue_circle:|src/util/coding.h|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 55 / 63 (87.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_text.cpp|1|5|20.00%|[389, 390, 391, 393]|
|:large_blue_circle:|src/util/compression/stream_compression.cpp|50|54|92.59%|[430, 480, 488, 500]|
|:large_blue_circle:|src/util/coding.h|4|4|100.00%|[]|

File: be/src/exec/hdfs_scanner_text.cpp [hunk 1]
```diff
@@ -385,6 +385,14 @@ Status HdfsTextScanner::_create_or_reinit_reader() {
 }
 
 Status HdfsTextScanner::_build_hive_column_name_2_index() {
```
Comment (imay) [inline be/src/exec/hdfs_scanner_text.cpp:387]: Could you let me know why you changed this function?
It seems this doesn't relate to the snappy compression.

Comment (imay) [reply be/src/exec/hdfs_scanner_text.cpp:387]: And better to give some comments for this newly added logic

Comment (dirtysalt) [reply be/src/exec/hdfs_scanner_text.cpp:387]: It's only for test. When we use file table for testing, there is no hive names. 

And when there is no hive names, we use slot orders. (And that's also the logic of orc scanner).

```
CREATE EXTERNAL TABLE test_csv_snappy_format
(
    a string,
    b string
) 
ENGINE=file
PROPERTIES 
(
    "path" = "${snappy_prefix[1]}",
    "format" = "text",
    "column_separator" = ","
);
```

Comment (dirtysalt) [reply be/src/exec/hdfs_scanner_text.cpp:387]: I'll add comment later.

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 55 / 63 (87.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_text.cpp|1|5|20.00%|[389, 390, 391, 393]|
|:large_blue_circle:|src/util/compression/stream_compression.cpp|50|54|92.59%|[430, 480, 488, 500]|
|:large_blue_circle:|src/util/coding.h|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 55 / 63 (87.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_text.cpp|1|5|20.00%|[391, 392, 393, 395]|
|:large_blue_circle:|src/util/compression/stream_compression.cpp|50|54|92.59%|[430, 480, 488, 500]|
|:large_blue_circle:|src/util/coding.h|4|4|100.00%|[]|

**Review by imay — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 55 / 63 (87.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_text.cpp|1|5|20.00%|[391, 392, 393, 395]|
|:large_blue_circle:|src/util/compression/stream_compression.cpp|50|54|92.59%|[430, 480, 488, 500]|
|:large_blue_circle:|src/util/coding.h|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 55 / 63 (87.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_text.cpp|1|5|20.00%|[391, 392, 393, 395]|
|:large_blue_circle:|src/util/compression/stream_compression.cpp|50|54|92.59%|[430, 480, 488, 500]|
|:large_blue_circle:|src/util/coding.h|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 55 / 63 (87.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_text.cpp|1|5|20.00%|[391, 392, 393, 395]|
|:large_blue_circle:|src/util/compression/stream_compression.cpp|50|54|92.59%|[430, 480, 488, 500]|
|:large_blue_circle:|src/util/coding.h|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 55 / 63 (87.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_text.cpp|1|5|20.00%|[391, 392, 393, 395]|
|:large_blue_circle:|src/util/compression/stream_compression.cpp|50|54|92.59%|[430, 480, 488, 500]|
|:large_blue_circle:|src/util/coding.h|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 55 / 63 (87.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_text.cpp|1|5|20.00%|[391, 392, 393, 395]|
|:large_blue_circle:|src/util/compression/stream_compression.cpp|50|54|92.59%|[430, 480, 488, 500]|
|:large_blue_circle:|src/util/coding.h|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 55 / 63 (87.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_text.cpp|1|5|20.00%|[391, 392, 393, 395]|
|:large_blue_circle:|src/util/compression/stream_compression.cpp|50|54|92.59%|[430, 480, 488, 500]|
|:large_blue_circle:|src/util/coding.h|4|4|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:heart_eyes: **pass** : 55 / 63 (87.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/hdfs_scanner_text.cpp|1|5|20.00%|[391, 392, 393, 395]|
|:large_blue_circle:|src/util/compression/stream_compression.cpp|50|54|92.59%|[430, 480, 488, 500]|
|:large_blue_circle:|src/util/coding.h|4|4|100.00%|[]|

**Review by letian-jiang — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by Smith-Cruise — APPROVED** [review]

**Review by zombee0 — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Comment by dirtysalt** [comment]
@mergify backport branch-3.1

**Comment by dirtysalt** [comment]
@mergify backport branch-3.2

**Comment by dirtysalt** [comment]
@mergify backport branch-3.2

**Pull request merged**